### PR TITLE
closeAll should attempt to close all ReadCloser's

### PR DESCRIPTION
**Pull request by tedyu**
closeAll() in wal.go should try to close all readers.

One of the closer error is returned.
The callers of closeAll() currently don't check the error return.

### CODE REVIEW

**Comment by tedyu** [comment]
@jingyih 
Can you take a look ?

**Comment by jingyih** [comment]
Thanks! This looks like a bug. We probably also want to pass in the logger to log error on each failed close.

cc @gyuho

**Comment by tedyu** [comment]
@jingyih 
Your comment has been addressed.

**Comment by tedyu** [comment]
@gyuho can you review this ?
Thanks

File: wal/wal.go [hunk 1]
```diff
@@ -886,11 +886,13 @@ func (w *WAL) seq() uint64 {
 	return seq
 }
 
-func closeAll(rcs ...io.ReadCloser) error {
+func closeAll(lg *zap.Logger, rcs ...io.ReadCloser) error {
+	var errRet error
```
Comment (gyuho) [inline wal/wal.go:890]: can we expose all errors? (collect them in slice, and return aggregated error at the end)?

File: wal/wal.go [hunk 2]
```diff
@@ -886,11 +886,13 @@ func (w *WAL) seq() uint64 {
...
 	for _, f := range rcs {
 		if err := f.Close(); err != nil {
-			return err
+			lg.Error("failed to close: " + err.Error())
```
Comment (gyuho) [inline wal/wal.go:893]: `lg.Warn("failed to close", zap.Error(err))`?

File: wal/wal.go (hunk 1 above)
Comment (tedyu) [reply wal/wal.go:890]: As I noted in the description, the return value is ignored by callers.

If you still feel aggregation is needed, @gyuho can you point me to an example where error slice is aggregated ?
I found errsToError() but it is in functional/tester/utils.go

Thanks

Comment (tedyu) [reply wal/wal.go:890]: @gyuho 
Waiting for your comment.

Comment (gyuho) [reply wal/wal.go:890]: I was thinking just aggregate as slice of strings, and join in return?

Comment (gyuho) [reply wal/wal.go:890]: `multierr` looks good as well :)

Comment (tedyu) [reply wal/wal.go]: @gyuho 
I adopted your first suggestion.

Please take a look.

File: wal/wal.go [hunk 3]
```diff
@@ -886,11 +887,13 @@ func (w *WAL) seq() uint64 {
...
+func closeAll(lg *zap.Logger, rcs ...io.ReadCloser) error {
+	stringArr := make([]string, 0)
 	for _, f := range rcs {
 		if err := f.Close(); err != nil {
-			return err
+			lg.Warn("failed to close: ", zap.Error(err))
+			stringArr = append(stringArr, err.Error())
 		}
 	}
-	return nil
```
Comment (gyuho) [inline wal/wal.go:895]:
```go
if len(stringArr) == 0 { return nil }
return errors.New(strings.Join(stringArr, ", "))
```
?

Comment (tedyu) [reply wal/wal.go:895]: @gyuho 
Addressed.

**Pull request merged**